### PR TITLE
feat(storage): Add support for dual region gcs buckets

### PR DESCRIPTION
**Pull request by SandeepTuniki**
Addresses ##18329 and #18330

### CODE REVIEW

**Comment by SandeepTuniki** [comment]
The flaky failing test is unrelated to this PR, and is being tracked at #18854

**Review by dazuma — COMMENTED** [review]
This seems like it's more than just `docs:`. There are functionality changes, a new `Bucket#data_locations` method, and a new `custom_placement_config` argument. Is that right?

File: google-cloud-storage/lib/google/cloud/storage/project.rb [hunk 1]
```diff
@@ -290,9 +290,9 @@ def bucket bucket_name,
         #     access, and allUsers get READER access.
         # @param [String] location The location of the bucket. Optional.
         #   If not passed, the default location, 'US', will be used.
-        #   If specifying a dual-region location, the regions can be specified
-        #   by joining them with a plus sign, for example 'US-CENTRAL1+US-WEST1'. See:
-        #   [Storage Locaitons](https://cloud.google.com/storage/docs/locations).
+        #   If specifying a dual-region location, the `customPlacementConfig`
+        #   property should be set in conjunction. See:
+        #   [Storage Locatons](https://cloud.google.com/storage/docs/locations).
```
Comment (bajajneha27) [inline google-cloud-storage/lib/google/cloud/storage/project.rb:295]: There's still a typo :P

**Review by bajajneha27 — COMMENTED** [review]
LGTM overall except for one minor comment.

File: google-cloud-storage/lib/google/cloud/storage/project.rb (hunk 1 above)
Comment (SandeepTuniki) [reply google-cloud-storage/lib/google/cloud/storage/project.rb]: Corrected 🙈

**Comment by SandeepTuniki** [comment]
> This seems like it's more than just `docs:`. There are functionality changes, a new `Bucket#data_locations` method, and a new `custom_placement_config` argument. Is that right?

@dazuma Yes, I think this should be `feat`. I considered this similar to #17997. But this change would be part of GA, so `feat` is probably correct. @danielbankhead Can you confirm please?

**Review by bajajneha27 — APPROVED** [review]

**Comment by danielbankhead** [comment]
@SandeepTuniki yes, should be `feat`

**Pull request merged**